File: app/src/main/java/com/example/android/popularmovies/view_model/MainViewModelFactory.java
package com.example.android.popularmovies.view_model;

import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;
import androidx.annotation.NonNull;

import com.example.android.popularmovies.repository.MovieRepository;


public class MainViewModelFactory implements ViewModelProvider.Factory {

    private MovieRepository repository;

    public MainViewModelFactory(MovieRepository repository) {
        this.repository = repository;
    }


    @NonNull
    @Override
    @SuppressWarnings("unchecked")
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new MainViewModel(repository);
    }
}

File: app/src/main/kotlin/com/example/android/popularmovies/view_model/MainViewModelFactory.kt
package com.example.android.popularmovies.view_model

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider


class MainViewModelFactory(private val mApiKey: String) : ViewModelProvider.Factory {
    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return MainViewModel(mApiKey) as T
    }
}
